### DISCUSSION TITLE
How to import MDX components in .tsx files?

### DISCUSSION THREAD

**Discussion by wuchendi** [contentlayerdev/contentlayer#514] [Q&A]
I am using contentlayer and MDX, and I am testing based on the example template [shadcn/next-contentlayer](https://github.com/shadcn/next-contentlayer).

However, I have encountered an issue and I'm not sure how to correctly import MDX components in .tsx files.

code:
![image](https://github.com/contentlayerdev/contentlayer/assets/31385662/1e632537-0d1e-4c05-a6e5-10583992966a)

**Comment by schardev** [comment]
You cannot just import `.mdx` files and expect them to work as component. You have to give the raw code to `useMDXComponent()` hook you get from `next-contentlayer/hooks` which will in turn give you a component that you can render. See [this](https://github.com/shadcn/next-contentlayer/blob/main/components/mdx-components.tsx) and [this](https://github.com/shadcn/next-contentlayer/blob/5b32397c2034a97a077cd695f3421a2e76af05c5/app/%5B...slug%5D/page.tsx#L57).

**Comment by wuchendi** [comment]
> You cannot just import `.mdx` files and expect them to work as component. You have to give the raw code to `useMDXComponent()` hook you get from `next-contentlayer/hooks` which will in turn give you a component that you can render. See [this](https://github.com/shadcn/next-contentlayer/blob/main/components/mdx-components.tsx) and [this](https://github.com/shadcn/next-contentlayer/blob/5b32397c2034a97a077cd695f3421a2e76af05c5/app/%5B...slug%5D/page.tsx#L57).


Thank you for your reply. Is that right?
![image](https://github.com/contentlayerdev/contentlayer/assets/31385662/d455762f-3222-460c-b4df-375c49a1de50)

**Comment by schardev** [comment]
@WuChenDi I will suggest you to go through the official contentlayer docs. Start with this [page](https://www.contentlayer.dev/docs/sources/files/mdx-d747e46d) if you have already set up most stuff.
